packages/engine-core/src/common/utils/runtimeHeadersToHttpHeaders.ts

```diff
@@ -0,0 +1,21 @@
+import { QueryEngineRequestHeaders } from '../types/QueryEngine'
+
+/**
+ * Takes runtime data headers and turns it into QE HTTP headers
+ * @param headers to transform
+ * @returns
+ */
+export function runtimeHeadersToHttpHeaders(headers: QueryEngineRequestHeaders): Record<string, string | undefined> {
+  return Object.keys(headers).reduce((acc, runtimeHeaderKey) => {
+    let httpHeaderKey = runtimeHeaderKey
+
+    if (runtimeHeaderKey === 'transactionId') {
+      httpHeaderKey = 'X-transaction-id'
+    }
+
+    // if header key isn't changed, a copy happens
+    acc[httpHeaderKey] = headers[runtimeHeaderKey]
+
+    return acc
+  }, {} as Record<string, string | undefined>)
```

Comment:
```suggestion
  if (headers.transactionId) {
      const { transactionId, ...httpHeaders } = headers
      httpHeaders['X-transaction-id'] = transactionId
      return httpHeaders
  }
  return headers
```
No need to traverse whole object for one rename.